app/services/cart_service.rb

```diff
@@ -143,11 +138,22 @@ def check_variant_available_under_distribution(variant)
     false
   end
 
-  def line_item_for_variant_id(variant_id)
-    order.find_line_item_by_variant Spree::Variant.find(variant_id)
+  def line_item_for_variant(variant)
+    order.find_line_item_by_variant variant
   end
 
   def variant_ids_in_cart
     @order.line_items.pluck :variant_id
   end
+
+  def indexed_variants(variants_data)
```

Comment:
this is only used in the private method attempt_cart_add_variants, I'd move this below that method to reduce scrolling up and down.